src/api/cpp/cvc5.cpp

```diff
@@ -6564,7 +6564,9 @@ Term Solver::simplify(const Term& term)
   CVC5_API_TRY_CATCH_BEGIN;
   CVC5_API_SOLVER_CHECK_TERM(term);
   //////// all checks before this line
-  return Term(this, d_slv->simplify(*term.d_node));
+  Term res = Term(this, d_slv->simplify(*term.d_node));
+  Assert(res.getSort() == term.getSort());
```

Comment:
Actually, until subtyping is eliminated, this needs to be relaxed since we allow Real to be simplified to Int.